src/status.rs

```diff
@@ -1,167 +1,162 @@
-use std::fmt;
-
+use bytes::Bytes;
 use h2;
-use http::header::{HeaderMap, HeaderValue};
-use http::status::StatusCode;
+use http::{self, HeaderMap};
+use http::header::{HeaderValue, InvalidHeaderValueBytes};
+use std::io;
 
 #[derive(Debug, Clone)]
 pub struct Status {
     code: Code,
+    error_message: Option<Bytes>,
```

Comment:
I think it would be better if `error_message` was a `String`. In this comment I argue for changing `Bytes` => `String`, in the comment below I argue for removing `Option`.

The error message field is supposed to be human readable. Storing them as `Bytes` makes this considerably more painful, just for handling the very rare corner case of non-UTF-8 error messages. Java and C# gRPC store this message as a (unicode) String. Java does percent encoding of invalid characters, [C# fails nongracefully to parse invalid UTF-8](https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core/Internal/AsyncCall.cs#L119) (ping @jtattermusch is this true?).

I think the pain of this not being a String vs the value of supporting arbitrary binary error messages is too large to justify `error_message` being `Bytes`.

https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core/Internal/BatchContextSafeHandle.cs#L87
https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/Status.java#L353

Reply:
Yeah, I went back and forth on this myself; I went with `Bytes` mostly because there isn't an obviously good place to complain that it was invalid if it was, but have made it a `String` because that's probably more useful.

Reply:
Thanks for updating the code!

Reply:
yes, the error message should be human-readable string (utf8 by default). If one wants to send additional error details that are binary, the right way is to add a binary-values trailer with extra debug info.  According to best practices, services can attach protobuf-serialized error status as an extra trailer. https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto